File: 백준/Gold/2230. 수 고르기/수 고르기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int[] numbers = new int[n];
        for (int i = 0; i < n; i++) {
            numbers[i] = Integer.parseInt(br.readLine());
        }

        Arrays.sort(numbers);
        
        int end = 0;
        int min = Integer.MAX_VALUE;
        for (int start = 0; start < n; start++) {
            while (end < n && numbers[end] - numbers[start] < m) end++;
            if (end == n) break;
            min = Math.min(min, numbers[end] - numbers[start]);
        }

        System.out.print(min);
    }
}

File: 백준/Gold/2230. 수 고르기/수 고르기.kt
import java.lang.Integer.min

fun main() {
    val (n, m) = readln().split(" ").map { it.toInt() }
    val list = ArrayList<Int>()
    for (i in 0 until n) {
        list.add(readln().toInt())
    }
    list.sort()
    var en = 0
    var min = 0x7fffffff
    for (st in 0 until n) {
        while (en < n && list[en] - list[st] < m) en++
        if (en == n) break
        min = min(min, list[en] - list[st])
    }
    println(min)
}